colony/branch_utils.py

```diff
@@ -97,7 +100,8 @@ def switch_to_temp_branch(repo: BlueprintRepo, defined_branch_in_file: str) -> s
     uncommitted_branch_name = UNCOMMITTED_BRANCH_NAME + defined_branch_in_file + "-" + random_suffix
     try:
         # todo return id and use it for revert_from_temp_branch
-        stash_local_changes_and_preserve_uncommitted_code(repo)
+        if repo.is_dirty() or repo.untracked_files:
```

Comment:
This looks like a bug no? untracked files are not stashed so we'll run into the same issue. 
We should only stash when we have uncommitted staged code

Reply:
they are stashed:
repo.git.stash("save", "--include-untracked")

Reply:
Now it`s part of logger information